B 29 feature with retouched photos

OleksandrGubariev opened the pull request.

OleksandrGubariev committed:
[b-29] Feature with retouched photos

OleksandrGubariev committed:
Add resize photo and watermark

OleksandrGubariev committed:
delete commented code

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -54,8 +54,9 @@
-    private static final String UPDATE_PAID_PHOTOS = "UPDATE OrderPhotos SET photoStatusId = 3 WHERE orderId = ? AND photoStatusId = 2";
     private static final String GET_PHOTOS_BY_STATUS_AND_ORDER_ID = "SELECT * FROM OrderPhotos WHERE orderId=? AND photoStatusId = ?";
+    private static final String GET_PHOTOS_SOURCES_BY_ORDER_ID = "SELECT * FROM OrderPhotos WHERE orderId=?";
+    private static final String UPDATE_RETOUCHED_PHOTOS_STATUS = "UPDATE OrderPhotos SET photoStatusId = 3 WHERE source = ?";
```
А если одно фото в нескольких заказах будет?

OleksandrGubariev replied:
Спасибо. Скорее одно и то же имя может быть. Добавлю еще условие orderId

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/entity/photo/Photos.java:
```diff
@@ -0,0 +1,24 @@
+    public void setRetouchedPhotosPath(List<String> retouchedPhotosPath) {
+        this.retouchedPhotosPath = retouchedPhotosPath;
+    }
+}
```
А для списков @Setter, @Getter не работает?

OleksandrGubariev replied:
Проверю

OleksandrGubariev replied:
done

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos:
1. add download one photo for user
2. change delete photo
3. change delete order
4. fix bug with Download Selected

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos: add opportunity download paid photo for admin in ready and selected order status

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos: add change from develop

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos:
1. add opportunity for admin delete photo in status SELECTED
2. retouched and new photos are added by different streams

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -54,8 +54,11 @@
-    private static final String UPDATE_PAID_PHOTOS = "UPDATE OrderPhotos SET photoStatusId = 3 WHERE orderId = ? AND photoStatusId = 2";
+    //private static final String UPDATE_PAID_PHOTOS = "UPDATE OrderPhotos SET photoStatusId = 3 WHERE orderId = ? AND photoStatusId = 2";
+
     private static final String GET_PHOTOS_BY_STATUS_AND_ORDER_ID = "SELECT * FROM OrderPhotos WHERE orderId=? AND photoStatusId = ?";
```
Комментарий

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -54,8 +54,11 @@
+
     private static final String GET_PHOTOS_BY_STATUS_AND_ORDER_ID = "SELECT * FROM OrderPhotos WHERE orderId=? AND photoStatusId = ?";
+    private static final String GET_SELECTED_PHOTOS_SOURCES_BY_ORDER_ID = "SELECT * FROM OrderPhotos WHERE orderId=? AND photoStatusId=2";
+    private static final String UPDATE_RETOUCHED_PHOTOS_STATUS = "UPDATE OrderPhotos SET photoStatusId = 3 WHERE source = ? AND orderId = ? AND photoStatusId=2";
```
GET_PHOTOS_BY_STATUS_AND_ORDER_ID с параметром можна было бы использовать

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -424,6 +427,28 @@ public void editOrderByUser(int orderId, String commentUser) {
+            PreparedStatement preparedStatement = connection.prepareStatement(GET_SELECTED_PHOTOS_SOURCES_BY_ORDER_ID)){
+            preparedStatement.setInt(1, orderId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            List<String> photosSources = new ArrayList<>();
```
resultset где закрывается?

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/dao/file/LocalDiskPhotoDao.java:
```diff
@@ -62,42 +90,91 @@ public void deleteByOrder(int orderId) {
-            log.error("No paid photos in order with id : {}", orderId);
-            throw new RuntimeException("No paid photos in order with id " + orderId);
+    public List<String> saveRetouchedPhotoByOrder(List<Part> photosParts, int orderId, List<String> photosSources) {
+        String orderPath = getPathToOrderDir(orderId);
```
имя параметра - selectedSources или что-то в этом ключе было бы понятнее здесь

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/dao/file/LocalDiskPhotoDao.java:
```diff
@@ -136,7 +213,35 @@ private String getFileName(Part part) {
+            log.info("Save resizing photo : {}", toPhoto);
+        } catch (IOException e) {
+            log.error("Cannot read photo to resize");
+            throw new RuntimeException("Cannot read photo to resize", e);
```
ошибка в лог не попадет

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -436,12 +461,29 @@ public void savePhotos(int orderId, List<String> photosPaths) {
+            try (Connection connection = dataSource.getConnection();
+                 PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_RETOUCHED_PHOTOS_STATUS)) {
+                preparedStatement.setString(1, pathToPhoto);
+                preparedStatement.setInt(2, orderId);
```
тут в цикле много раз будет создаваться соединение, готовиться тот же запрос к БД, а нельзя ли использовать одно соединение и один PreparedStatement?

OleksandrGubariev replied:
done

Rooman replied:
батч...

Yuliia-Sh commented on src/main/java/com/photostudio/dao/jdbc/JdbcOrderDao.java:
```diff
@@ -436,12 +461,29 @@ public void savePhotos(int orderId, List<String> photosPaths) {
+                log.info("Photo :{} status is retouched", pathToPhoto);
+            } catch (SQLException e) {
+                log.error("Error during change status photo : {}", pathToPhoto);
+                throw new RuntimeException("Error during change status photo "+pathToPhoto, e);
```
в логе мы не увидем ошибки

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/web/filter/WatermarkFilter.java:
```diff
@@ -19,7 +19,7 @@
 public class WatermarkFilter implements Filter {
 
     @Override
```
Watermark фильтр нам получается вообще больше не нужен?

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/java/com/photostudio/web/servlet/DownloadPaidPhotosServlet.java:
```diff
@@ -0,0 +1,68 @@
+        String uri = request.getRequestURI();
+        String[] partsOfUri = uri.split("/");
+        int orderId = Integer.parseInt(partsOfUri[partsOfUri.length - 1]);
+        String stringPhotoId = request.getParameter("downloadPhotoId");
```
в UtilClass есть функция getIdFromPath

OleksandrGubariev replied:
done

Yuliia-Sh commented on src/main/webapp/WEB-INF/templates/order.html:
```diff
@@ -189,29 +190,44 @@ <h2 class="error" align="center" th:if="${errorMessage!=null}" th:text="${errorM
                         id="btnUserReselect"
-                        onclick="btnUserSubmit.disabled=false; disabled=true;"
+                        onclick="btnUserReselect.disabled=false; disabled=true; frmNew.submit()"
                 >
```
лучше верни как было

OleksandrGubariev replied:
done

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos: fix bugs

Yuliia-Sh committed:
Merge branch 'develop' into b_29_Feature_with_retouched_photos

OleksandrGubariev committed:
[b-29] Feature_with_retouched_photos: add system variable in PropertyReader

OleksandrGubariev committed:
Merge branch 'b_29_Feature_with_retouched_photos' of https://github.com/Rooman/PhotoStudio into b_29_Feature_with_retouched_photos
